roles/openshift_preflight/common/tasks/main.yml

```diff
@@ -0,0 +1,22 @@
+---
+# check content available on all hosts
+- when:
+    - not containerized
+  block:
+
+    - name: determine if yum update will work
+      action: check_yum_update
+      register: r
+
+    - set_fact:
+        oo_preflight_check_results: "{{ oo_preflight_check_results + [r|combine({'_task': 'check content available on all hosts'})] }}"
```

Comment:
> I'm quite new to Ansible. I feel it is hard to use the word reliable and this type of "code" together, it feels like a hack to me :)
There are many ways to introduce subtle mistakes.

My concern here is that this is relying on undocumented behavior by relying on redefining a variable that already exists.

> From a programming stand point, I think the current approach is already quite crazy and hard to maintain... But was the best we've got working with Ansible so far, after experimenting with different things.

I agree, it is definitely not the ideal solution to the problem.

> Having different variable names for each role means that introducing a new check requires changes in at least two different places -- the new role and changing the code that checks these variables at the end of the playbook run. And it would be very very easy to miss this second part and assume things are working.

I think I see the issue here. I'm assuming that the only change would need to be in the new role, and that the check at the end of the playbook would simply be looking for the existence of any errors in the known variables. The only case I see there needing to be a change at the end of the playbook for verification is when a new role is added for checks, which would still require changes to the playbook for applying that role. Since the callback plugin is responsible for tracking the actual warnings/errors. I'm thinking something like this for a test (assuming the modules returned the number of errors found):

role a:
```
openshift_preflight_a_error_count = "{{ result_1.errors + result_2.errors + result_3.errors }}"
```

role b:
```
openshift_preflight_b_error_count = "{{ result_1.errors + result_2.errors }}"
```

role c:
```
openshift_preflight_c_error_count = "{{ result_1.errors + result_2.errors + result_3.errors + result_4.errors }}"
```

preflight playbook:
```
...
- hosts: ...
  roles:
  - a
  - b
  - c
  post_tasks:
  - fail:
    msg: "Preflight errors exist"
    when: "{{ (openshift_preflight_a_error_count + openshift_preflight_b_error_count + openshift_preflight_c_error_count) > 0 }}"
```